keps/sig-node/4569-cgroup-v1-deprecation/kep.yaml

```diff
@@ -0,0 +1,37 @@
+title: Deprecate Support for cgroup v1 in Kubernetes
+kep-number: 4569
+authors:
+  - "@harche"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+status: implementable
+creation-date: 2024-04-04
+reviewers:
+  - "@mrunalp"
+  - "@SergeyKanzhelev"
+approvers:
+  - "@derekwaynecarr"
+  - "@dchen1107"
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.31"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.31"
+  beta: "TBD"
+  stable: "TBD"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: DeprecatedCgroupV1
```

Comment:
let's call it `RemovedCgroupV1`

Reply:
Changed the feature name to `RemovedCgroupV1`

Reply:
We could actually call it `CGroupV1`, _insta-deprecate_ the feature gate, but default it to true (so the feature is still available). Then later, leave it deprecated and change the default so you have to opt in.

I'd do something like that.

`RemovedCgroupV1` isn't a feature; support for version 1 control group management is the feature that we're changing (by eventually removing it?).

Reply:
I got the inspiration for using a feature gate for deprecation from this KEP https://github.com/kubernetes/enhancements/blob/e8045cfb4232ea9e17176eadc6fc67ca51fe8ba0/keps/sig-network/4004-deprecate-kube-proxy-version/README.md#goals

Reply:
Using a feature gate yes, but using `Removed` as the prefix doesn't feel right (and sets the wrong kind of precedent).

How about plain `CgroupV1`, with no alpha or beta, just deprecated-and-active-by-default followed by still-deprecated-and-disabled-by-default followed by: no more support, sorry folks.